flink-elastic-catalog/src/test/java/com/getindata/flink/connector/jdbc/catalog/ElasticCatalogTestBase.java

```diff
@@ -0,0 +1,130 @@
+
+import okhttp3.Credentials;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.apache.commons.compress.utils.IOUtils;
+import org.junit.BeforeClass;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertTrue;
+
+class ElasticCatalogTestBase {
+    protected static ElasticJdbcCatalogFactory catalogFactory;
+    protected static String url;
+    protected static final ElasticsearchTestContainer container = new ElasticsearchTestContainer();
+    protected static final String USERNAME = "elastic";
+    protected static final String PASSWORD = "password";
+
+    private static final int REQUEST_RETRY_MAX_COUNT = 3;
+    private static final int TIMEOUT_IN_SECONDS = 3;
+
+    @BeforeClass
+    public static void beforeAll() throws Exception {
+        container.withEnv("xpack.security.enabled", "true");
+        container.withEnv("ELASTIC_PASSWORD", PASSWORD);
+        container.withEnv("ES_JAVA_OPTS", "-Xms1g -Xmx1g");
+        container.start();
+        TimeUnit.SECONDS.sleep(TIMEOUT_IN_SECONDS);
+        if (!isTrialEnabled()) {
+            enableTrial();
+        }
+        catalogFactory = new ElasticJdbcCatalogFactory();
+
+        url = String.format("jdbc:elasticsearch://%s:%d", container.getHost(),
+                container.getElasticPort());
+    }
+
+    private static void enableTrial() throws Exception {
+        OkHttpClient client = new OkHttpClient();
+        Request request = new Request.Builder()
+                .url(String.format("http://%s:%d/_license/start_trial?acknowledge=true",
+                        container.getHost(), container.getElasticPort()))
+                .post(RequestBody.create(new byte[]{}))
+                .addHeader("Authorization", Credentials.basic(USERNAME, PASSWORD))
+                .build();
+        try (Response response = client.newCall(request).execute()) {
+            assertTrue(response.isSuccessful());
+        }
+    }
+
+    private static boolean isTrialEnabled() throws Exception {
+        OkHttpClient client = new OkHttpClient();
+        Request request = new Request.Builder()
+                .url(String.format("http://%s:%d/_license",
+                        container.getHost(), container.getElasticPort()))
+                .addHeader("Authorization", Credentials.basic(USERNAME, PASSWORD))
+                .build();
+        for (int i = 0; i < REQUEST_RETRY_MAX_COUNT; i++) {
+            Response response = client.newCall(request).execute();
+            if (response.code() == 200) {
+                return response.body().string().contains("\"type\" : \"trial\"");
+            }
+            TimeUnit.SECONDS.sleep(TIMEOUT_IN_SECONDS);
+        }
+        return false;
```

Comment:
If we never get 200, throw exception instead of returning false.

Reply:
Yeah was thinking about that :)